### PR TITLE
Introduce ability to use Filesystem service presigned URLs to upload/download model artifacts

### PR DESCRIPTION
**Pull request by artjen**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/artjen/mlflow/pull/11534?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11534/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11534
```

</p>
</details>

### What changes are proposed in this pull request?
Introduce the ability to upload and download model artifacts using presigned URLs provided by an API. This is guarded behind the `MLFLOW_UNITY_CATALOG_PRESIGNED_URLS_ENABLED` environment variable. 

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py [hunk 1]
```diff
@@ -1500,3 +1503,38 @@ def test_store_ignores_hive_metastore_default_from_spark_session(mock_http, spar
...
+
+
+def test_store_use_presigned_url_store_when_enabled(monkeypatch):
+    monkeypatch.setenv(MLFLOW_UNITY_CATALOG_PRESIGNED_URLS_ENABLED.name, "True")
+    with mock.patch("mlflow.utils.databricks_utils.get_config"):
+        uc_store = UcModelRegistryStore(store_uri="databricks-uc", tracking_uri="databricks-uc")
+        model_version = ModelVersion(name="catalog.schema.model_1", version="1")
+        presigned_store = uc_store._get_artifact_repo(model_version)
+
+        assert type(presigned_store) is PresignedUrlArtifactRepository
```
Comment (kriscon-db) [inline tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py:1540]: can we assert the type in the other test above as well?  Same with the 2nd time we test this down below.

**Comment by kriscon-db** [comment]
Overall LGTM but let's get someone from OSS team to take a look at it as well.

**Comment by BenWilson2** [comment]
Is there any need to support the chunked and chunk-based retry logic for multi-part downloads and uploads for large models within this artifact store implementation? 
Is that future work that is planned, or has that been deemed to be not needed for UC?

**Comment by artjen** [comment]
> Is there any need to support the chunked and chunk-based retry logic for multi-part downloads and uploads for large models within this artifact store implementation? Is that future work that is planned, or has that been deemed to be not needed for UC?

@BenWilson2 chunked and chunk-based retry logic for multi-part downloads and uploads for large models files will be needed and is planned as future work.

**Comment by BenWilson2** [comment]
Sounds great! I think the implementation looks fantastic!

**Review by BenWilson2 — APPROVED** [review]
LGTM!

**Comment by artjen** [comment]
> Sounds great! I think the implementation looks fantastic!

@BenWilson2 Thanks! Could you approve the PR if it looks good to you?

**Review by kriscon-db — APPROVED** [review]

**Pull request merged**